analyzers/src/SonarAnalyzer.CSharp/Rules/ImplementIDisposableCorrectly.cs

```diff
@@ -269,15 +268,20 @@ private void VerifyDispose(MethodDeclarationSyntax disposeMethod, bool isSealedC
                 typeSymbol.GetMembers()
                           .OfType<IMethodSymbol>()
                           .Where(predicate)
-                          .SelectMany(m => m.DeclaringSyntaxReferences)
-                          .Select(r => r.GetSyntax());
-
-            private static bool HasVirtualDisposeBool(INamedTypeSymbol symbol) =>
-                symbol.GetSelfAndBaseTypes()
-                      .SelectMany(t => t.GetMembers())
-                      .OfType<IMethodSymbol>()
-                      .Where(IsDisposeBool)
-                      .Any(methodSym => !methodSym.IsAbstract);
+                          .SelectMany(symbol => symbol.PartialImplementationPart?.DeclaringSyntaxReferences ?? symbol.DeclaringSyntaxReferences)
+                          .Select(reference => reference.GetSyntax());
+
+            private static IEnumerable<SyntaxNode> FindMethodImplementation(INamedTypeSymbol typeSymbol, Func<IMethodSymbol, bool> predicate, TypeDeclarationSyntax typeDeclarationSyntax) =>
```

Comment:
this method returns both implementations (partial or not) and abstract methods (so not implemented). should improve the name. e.g. `FindMethodImplementationOrAbstractDeclaration` (or better)

Would be useful to document why we look also for abstract methods.